xarray/tests/test_backends.py

```diff
@@ -351,6 +351,16 @@ def test_dtype_coercion_error(self) -> None:
                 with pytest.raises(ValueError, match="could not safely cast"):
                     ds.to_netcdf(path, format=format)
 
+    def test_literal_timedelta_fill_value_coercion_error(self) -> None:
+        for format in self.netcdf3_formats:
+            timedeltas = np.array(
+                [0, np.iinfo(np.int32).min, np.iinfo(np.int64).min]
+            ).astype("timedelta64[s]")
```

Comment:
I would suggest doing this test with `NaT` as well (or instead)

Reply:
Indeed this test covers that since `np.iinfo(np.int64).min` is `NaT`.  I'll write the constructor like this instead so that becomes clearer:
```
timedeltas = np.array([0, np.iinfo(np.int32).min, "NaT"], dtype="timedelta64[s]")
```